### PR TITLE
add custom logger to presigned_url function

### PR DESCRIPTION
**Pull request by bogdan-mania-moj**

### CODE REVIEW

File: containers/daap-presigned-url/src/var/task/presigned_url.py [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+logger = DataPlatformLogger(
+    extra={
+        "image_version": os.getenv("VERSION", "unknown"),
+        "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+    }
+)
+
 
 def handler(event, context):
     bucket_name = os.environ["BUCKET_NAME"]
```
Comment (murdo-moj) [inline containers/daap-presigned-url/src/var/task/presigned_url.py:25]: after L20 you can log inputs from queryStringParameters aka the only ones which will change from execution to executipn

File: containers/daap-authorizer/src/var/task/authorizer.py [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+    extra={
+        "image_version": os.getenv("VERSION", "unknown"),
+        "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+    }
+)
+
 
 def handler(event, context):
-    logging.info(f"event: {event}")
+    logger.info(f"event: {event}")
```
Comment (LavMatt) [inline containers/daap-authorizer/src/var/task/authorizer.py:17]: can add in the lambda function name to the logger inside the handler.
```suggestion

    logger.add_extras({"lambda_name": context.function_name})

File: containers/daap-get-glue-metadata/src/var/task/get_glue_metadata.py [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+    }
+)
 
 
 def handler(event, context):
     database = event["queryStringParameters"]["database"]
     table = event["queryStringParameters"]["table"]
 
+    logger.info(f"database: {database}")
+    logger.info(f"table: {table}")
```
Comment (LavMatt) [inline containers/daap-get-glue-metadata/src/var/task/get_glue_metadata.py]: should use this instead
```suggestion
    logger.add_extras(
        {
            "lambda_name": context.function_name,
            "data_product_name": database,
            "table_name": table
        }
    )

File: containers/daap-presigned-url/src/var/task/presigned_url.py [hunk 4]
```diff
@@ -58,7 +63,13 @@ def handler(event, context):
         ["content-length-range", 0, 5000000000],
     ]
```
Comment (LavMatt) [inline containers/daap-presigned-url/src/var/task/presigned_url.py:70]:
```suggestion

    logger.add_extras(
        {
            "lambda_name": context.function_name,
            "data_product_name": database,
            "table_name": table
        }
    )

File: containers/daap-presigned-url/src/var/task/presigned_url.py [hunk 5]
```diff
@@ -58,7 +63,13 @@ def handler(event, context):
...
 
-    root_logger.info(f"s3 key: {file_name}")
+    logger.info(f"s3 bucket: {bucket_name}")
+    logger.info(f"s3 key: {file_name}")
+    logger.info(f"database: {database}")
+    logger.info(f"table: {table}")
```
Comment (LavMatt) [inline containers/daap-presigned-url/src/var/task/presigned_url.py:82]:
```suggestion

**Review by LavMatt — COMMENTED** [review]
Have added a few suggestions. There is some extra info that can be added to the logger inside the handler.

Could you also please write the log to s3 as json? There is a method of the logger to do this, just need bucket name and security params as [here](https://github.com/ministryofjustice/data-platform/blob/main/containers/daap-athena-load/src/var/task/athena_load_handler.py#L22-L25) and [here](https://github.com/ministryofjustice/data-platform/blob/main/containers/daap-athena-load/src/var/task/athena_load_handler.py#L77C1-L77C70)

File: containers/daap-authorizer/src/var/task/authorizer.py [hunk 6]
```diff
@@ -1,10 +1,19 @@
...
 
+from data_platform_logging import DataPlatformLogger
+
+logger = DataPlatformLogger(
+    extra={
+        "image_version": os.getenv("VERSION", "unknown"),
+        "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+    }
+)
+
```
Comment (LavMatt) [inline containers/daap-authorizer/src/var/task/authorizer.py:12]:
```suggestion

s3_security_opts = {
    "ACL": "bucket-owner-full-control",
    "ServerSideEncryption": "AES256",
}

File: containers/daap-authorizer/src/var/task/authorizer.py [hunk 7]
```diff
@@ -1,10 +1,19 @@
...
 
 def handler(event, context):
-    logging.info(f"event: {event}")
+    logger.info(f"event: {event}")
+    logger.add_extras({"lambda_name": context.function_name})
```
Comment (LavMatt) [inline containers/daap-authorizer/src/var/task/authorizer.py:16]:
```suggestion
    logger.add_extras({"lambda_name": context.function_name})
    logger.info(f"event: {event}")
    logger.write_log_dict_to_s3_json(bucket_name, **s3_security_opts)

File: containers/daap-get-glue-metadata/src/var/task/get_glue_metadata.py [hunk 8]
```diff
@@ -1,12 +1,29 @@
...
 
 import boto3
+from data_platform_logging import DataPlatformLogger
+
+logger = DataPlatformLogger(
+    extra={
+        "image_version": os.getenv("VERSION", "unknown"),
+        "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+    }
+)
```
Comment (LavMatt) [inline containers/daap-get-glue-metadata/src/var/task/get_glue_metadata.py:13]:
```suggestion
s3_security_opts = {
    "ACL": "bucket-owner-full-control",
    "ServerSideEncryption": "AES256",
}

File: containers/daap-get-glue-metadata/src/var/task/get_glue_metadata.py [hunk 9]
```diff
@@ -1,12 +1,29 @@
...
     table = event["queryStringParameters"]["table"]
 
+    logger.add_extras(
+        {
+            "lambda_name": context.function_name,
+            "data_product_name": database,
+            "table_name": table,
+        }
+    )
+
```
Comment (LavMatt) [inline containers/daap-get-glue-metadata/src/var/task/get_glue_metadata.py:26]:
```suggestion
    logger.info(f"event: {event}")
    logger.write_log_dict_to_s3_json(bucket_name, **s3_security_opts)

**Review by LavMatt — COMMENTED** [review]
Can handle write to s3 of log file when common path module is available

**Review by LavMatt — APPROVED** [review]

**Review by julialawrence — APPROVED** [review]

**Pull request merged**